pkg/cache/context.go

```diff
@@ -454,6 +454,41 @@ func (ctx *Context) IsPodFitNode(name, node string, allocate bool) error {
 	return fmt.Errorf("predicates were not running because pod or node was not found in cache")
 }
 
+func (ctx *Context) IsPodFitNodeViaPreemption(name, node string, allocations []string, startIndex int) (index int, ok bool) {
```

Comment:
Please add unit test coverage in context_test.go.

Reply:
This is going to be virtually impossible for the same reasons we also don't have coverage for `IsPodFitNode()` as doing so would require mocking essentially the entire stack underneath. This is why we will do this via e2e tests once the core is complete.